packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1717,9 +1735,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                     referenceSequenceNumber: summaryRefSeqNum,
                 };
 
-            const handle = await this.storage.uploadSummaryWithContext(
-                summarizeResult.summary,
-                summaryContext);
+            let handle: string;
+            try {
+                handle = await this.storage.uploadSummaryWithContext(summarizeResult.summary, summaryContext);
+            } catch (error) {
+                return { ...attemptData, ...generateData, error };
```

Comment:
I wonder if there is better way to deal with errors. I.e. have caller pass in object on which this function sets properties as they become available (all props are optional). And then remove all try/catch like this, leaving it up to the caller to have single try/catch, examine that object with optional props and do logging.

Reply:
I thought about doing this... it would go hand-in-hand with the potential feature of adding properties along the way to performance events within execute/executeAsync.

Reply:
I looked at doing this a little bit- it will make the code slimmer, so I think probably it's worth it, but I want to punt it out of this PR. It's too big and that has additional changes to keep the strict typings.